src/history.rs

```diff
@@ -0,0 +1,140 @@
+use std::{collections::HashMap, fs, path::PathBuf};
+
+use bitcode::{Decode, Encode};
+use dirs::config_dir;
+
+use crate::WrappedErr;
+
+#[derive(Decode, Encode, Default)]
+pub struct DocumentHistory {
+	pub last_pages_opened: HashMap<String, usize>
+}
+
+impl DocumentHistory {
+	pub fn load() -> Result<Self, WrappedErr> {
+		let path = Self::history_path()?;
+		let data = fs::read(path)
+			.map_err(|e| WrappedErr(format!("Failed to read history file: {e}").into()))?;
+		bitcode::decode(&data)
+			.map_err(|e| WrappedErr(format!("Failed to decode history file: {e}").into()))
+	}
+
+	pub fn save(&self) -> Result<(), WrappedErr> {
+		let path = Self::history_path()?;
+		fs::write(path, bitcode::encode(self))
+			.map_err(|e| WrappedErr(format!("Failed to write history file: {e}").into()))?;
+		Ok(())
+	}
+
+	fn history_path() -> Result<PathBuf, WrappedErr> {
+		config_dir()
+			.map(|p| p.join("tdf.history.bin"))
+			.ok_or_else(|| WrappedErr("Could not determine history directory".into()))
+	}
+}
+
+#[cfg(test)]
```

Comment:
I appreciate you adding tests - we don't have much tests in this repo (I'm mostly relying on the benchmarks' ability to run as tests, but that doesn't really verify any correctness - it just verifies that nothing crashes), so this is a good step in the right direction.